lang/cs/Org.Apache.REEF.Wake/Remote/Impl/TransportServer.cs

```diff
@@ -89,10 +80,45 @@ public IPEndPoint LocalEndpoint
         /// </summary>
         public void Run()
         {
-            _listener.Start();
+            if (LocalEndpoint.Port == 0)
+            {
+                FindAPortAndStartListener();
+            }
+            else
+            {
+                _listener.Start();
```

Comment:
listener is allocated and assigned in the constructor first, and Allocated and assigned as necessary. If allocation fails, CLR runtime will throw and exception